### DISCUSSION TITLE
Export files with Image name

### DISCUSSION THREAD

**Discussion by hittypitty** [joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script#5] [Q&A] [answered]
Can we export files with image names instead of mockup names?

**Comment by joonaspaakko** [comment, marked as answer]
I assume you mean the input files? If so, the answer is no. 

It would technically be possible to implement in the script, but it was left out because the script supports mockups with multiple target smart objects, like 2 or more mobile device screen for instance. 

For example: If you have a mockup with 3 device screens. Which of the 3 input files would set the name for the output file? It didn't seem like a feature worth implementing because it doesn't work in all situations. 

You could possibly get around it using a batch renaming application, but I'm not sure how viable that is.

**Reply by joonaspaakko** [reply]
@hittypitty, there's now a new version that introduced a new output `filename` keyword `@input` that takes the input filename from the Smart Object target with the most input files. Example settings script here: https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script/issues/6#issuecomment-844494033
